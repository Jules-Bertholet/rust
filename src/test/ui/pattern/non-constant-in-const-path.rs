// Checks if we emit `PatternError`s correctly.
// This is also a regression test for #27895 and #68394.

static FOO: u8 = 10;

fn main() {
    let x = 0;
    let 0u8..=x = 0;
    //~^ ERROR: runtime values cannot be referenced in patterns
    let 0u8..=FOO = 0;
    //~^ ERROR: referencing statics in patterns is experimental
    match 1 {
        0 ..= x => {}
        //~^ ERROR: runtime values cannot be referenced in patterns
        0 ..= FOO => {}
        //~^ ERROR: referencing statics in patterns is experimental
    };
}
