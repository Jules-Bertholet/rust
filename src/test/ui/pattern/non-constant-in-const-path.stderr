error[E0080]: runtime values cannot be referenced in patterns
  --> $DIR/non-constant-in-const-path.rs:8:15
   |
LL |     let 0u8..=x = 0;
   |               ^

error[E0658]: referencing statics in patterns is experimental
  --> $DIR/non-constant-in-const-path.rs:10:15
   |
LL |     let 0u8..=FOO = 0;
   |               ^^^
   |
   = help: add `#![feature(static_in_pattern)]` to the crate attributes to enable

error[E0080]: runtime values cannot be referenced in patterns
  --> $DIR/non-constant-in-const-path.rs:13:15
   |
LL |         0 ..= x => {}
   |               ^

error[E0658]: referencing statics in patterns is experimental
  --> $DIR/non-constant-in-const-path.rs:15:15
   |
LL |         0 ..= FOO => {}
   |               ^^^
   |
   = help: add `#![feature(static_in_pattern)]` to the crate attributes to enable

error: aborting due to 4 previous errors

Some errors have detailed explanations: E0080, E0658.
For more information about an error, try `rustc --explain E0080`.
